packages/cli/src/lib/push/requiredKeyValidator.ts

```diff
@@ -0,0 +1,60 @@
+import {SchemaValue} from '@coveo/bueno';
+import {CaseInsensitiveDocument} from './caseInsensitiveDocument';
+
+export class RequiredKeyValidator<T> {
```

Comment:
Used to validate that some properties are present(using bueno).

When they are not present, bueno will throw an error with a meaningful message (eg: document XYZ has no documentID, or documentXYZ has a documentID that is not a valid URL, or documentXYZ has a title that is not a string, etc.)